internal/server/v2/observation/calculation.go

```diff
@@ -17,56 +17,137 @@ package observation
 
 import (
 	"context"
+	"net/http"
 
+	pb "github.com/datacommonsorg/mixer/internal/proto"
 	pbv2 "github.com/datacommonsorg/mixer/internal/proto/v2"
 	"github.com/datacommonsorg/mixer/internal/server/cache"
+	"github.com/datacommonsorg/mixer/internal/server/resource"
 	"github.com/datacommonsorg/mixer/internal/store"
 )
 
-// CalculateObservationResponses returns a list of ObservationResponses based
-// on StatisticalCalculation formulas for the missing data in an input
-// ObservationResponse.
-func CalculateObservationResponses(
+// Computes a calculation for a variable and entity, based on a formula and input data.
+func Calculate(
 	ctx context.Context,
 	store *store.Store,
-	inputResp *pbv2.ObservationResponse,
 	cachedata *cache.Cache,
-) []*pbv2.ObservationResponse {
-	calculatedResponses := []*pbv2.ObservationResponse{}
-	for variable, variableObservation := range inputResp.ByVariable {
+	metadata *resource.Metadata,
+	httpClient *http.Client,
+	formula string,
+	variable string,
```

Comment:
why need `variable`?

Reply:
It's just for constructing the final response (since the formula doesn't reference the target variable anywhere)

Reply:
maybe have a class to wrap the formula and the target variable together?